Test Numpy memory handler.
This repo shows how to reproduce a segmentation fault by calling numpy `PyDataMem_GetHandler`.

Steps to reproduce:
- Clone this repo
- `pip install "numpy==2.3.0" "Cython==3.1.3" "pytest" "meson==1.8.3" "meson-python>=0.16.0" "ninja==1.13.0"`
- ```
  pip install --editable . \
      --verbose --no-build-isolation \
      --config-settings editable-verbose=true
  ```
- `pytest tests/test_numpy_mem_handler.py`

Solution is to add to numpy_mem_handler.pyx
```python
cimport numpy as cnp
cnp.import_array()
````
See https://github.com/numpy/numpy/issues/29600.